### PR TITLE
Move remoting infra to Remote.Workspaces

### PR DESCRIPTION
**Pull request by tmat**
Adds a layer between Features and VisualStudio that implements remoting infrastructure (such as remote service invocations and solution snapshot mirroring).

This change will enable us to take dependency on IServiceBroker (in this new layer) and invoke our remote services on VS4Mac.

This will result in an additional assembly load in common RPS scenarios.

Blocked on
- [x] VSUnitTesting change: https://devdiv.visualstudio.com/DevDiv/_git/VSUnitTesting/pullrequest/264664
- [x] RPS exception: https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1168435/

### CODE REVIEW

**Comment by tmat** [comment]
@CyrusNajmabadi @sharwell @jasonmalinowski @dotnet/roslyn-ide PTAL

File: src/EditorFeatures/CSharpTest/Squiggles/ErrorSquiggleProducerTests.cs [hunk 1]
```diff
@@ -49,7 +49,7 @@ public async Task ErrorTagGeneratedForError()
             Assert.Equal(PredefinedErrorTypeNames.SyntaxError, firstSpan.Tag.ErrorType);
         }
 
-        [WpfFact, Trait(Traits.Feature, Traits.Features.ErrorSquiggles)]
+        [WpfFact(Skip = "https://github.com/dotnet/roslyn/issues/46463"), Trait(Traits.Feature, Traits.Features.ErrorSquiggles)]
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/CSharpTest/Squiggles/ErrorSquiggleProducerTests.cs:52]: still need to skip?  the linked issue is closed.

File: src/VisualStudio/Core/Def/Implementation/Watson/WatsonReporter.cs [hunk 2]
```diff
@@ -43,10 +42,36 @@ public static void InitializeFatalErrorHandlers(TelemetrySession session)
...
+            {
+                s_telemetrySessions = s_telemetrySessions.Remove(session);
+            }
+        }
+
+        public static void RegisterLogger(TraceSource logger)
+        {
+            lock (_guard)
+            {
+                s_loggers.Add(logger);
```
Comment (CyrusNajmabadi) [inline src/VisualStudio/Core/Def/Implementation/Watson/WatsonReporter.cs:65]: s_loggers is an ImmutableArray.  so this doesn't do anything (afaict).

Comment (tmat) [reply src/VisualStudio/Core/Def/Implementation/Watson/WatsonReporter.cs:65]: good catch

File: src/EditorFeatures/CSharpTest/Squiggles/ErrorSquiggleProducerTests.cs (hunk 1 above)
Comment (tmat) [reply src/EditorFeatures/CSharpTest/Squiggles/ErrorSquiggleProducerTests.cs:52]: Removed.

Comment (CyrusNajmabadi) [reply src/EditorFeatures/CSharpTest/Squiggles/ErrorSquiggleProducerTests.cs:52]: still seems relevant.

File: src/VisualStudio/Core/Def/Implementation/Watson/WatsonReporter.cs (hunk 2 above)
Comment (CyrusNajmabadi) [reply src/VisualStudio/Core/Def/Implementation/Watson/WatsonReporter.cs:65]: still seems to apply.

File: src/VisualStudio/Core/Test.Next/Remote/SerializationValidator.cs [hunk 3]
```diff
@@ -31,13 +34,36 @@ public override Task<T> GetAssetAsync<T>(Checksum checksum, CancellationToken ca
...
+            {
+                // using .Result here since we don't want to convert all calls to this to async.
+                // and none of ChecksumWithChildren actually use async
+                Children = ImmutableArray.CreateRange(collection.Select(c => validator.GetValueAsync<T>(c).Result));
+            }
+
+            public int Count => Children.Length;
+            public T this[int index] => Children[index];
+            IEnumerator IEnumerable.GetEnumerator() => GetEnumerator();
+            public IEnumerator<T> GetEnumerator() => Children.Select(t => t).GetEnumerator();
```
Comment (CyrusNajmabadi) [inline src/VisualStudio/Core/Test.Next/Remote/SerializationValidator.cs:52]: the .Select(t => t) seems very weird.

**Pull request merged**